Sample will now use Objectify

lesv opened the pull request:
Please take a look @tmatsuo

Commit:
Sample will now use Objectify

ludoch commented:
I see deleted Eclipse launch config files...Let me know how easily you can
run or deploy via Eclipse actions..

On Thu, May 14, 2015 at 6:46 PM, Les Vogel notifications@github.com wrote:

> ## Please take a look @tmatsuo https://github.com/tmatsuo
> 
> You can view, comment on, or merge this pull request online at:
> 
> https://github.com/GoogleCloudPlatform/appengine-maven-archetypes-java/pull/1
> Commit Summary
> - Sample will now use Objectify
> 
> File Changes
> - _M_ appengine-skeleton-archetype/pom.xml
>   https://github.com/GoogleCloudPlatform/appengine-maven-archetypes-java/pull/1/files#diff-0
>   (6)
> - _M_
>   appengine-skeleton-archetype/src/main/resources/META-INF/maven/archetype-metadata.xml
>   https://github.com/GoogleCloudPlatform/appengine-maven-archetypes-java/pull/1/files#diff-1
>   (2)
> - _M_
>   appengine-skeleton-archetype/src/test/resources/projects/basic/archetype.properties
>   https://github.com/GoogleCloudPlatform/appengine-maven-archetypes-java/pull/1/files#diff-2
>   (2)
> - _M_ guestbook-archetype/pom.xml
>   https://github.com/GoogleCloudPlatform/appengine-maven-archetypes-java/pull/1/files#diff-3
>   (6)
> - _M_
>   guestbook-archetype/src/main/resources/META-INF/maven/archetype-metadata.xml
>   https://github.com/GoogleCloudPlatform/appengine-maven-archetypes-java/pull/1/files#diff-4
>   (10)
> - _D_
>   guestbook-archetype/src/main/resources/archetype-resources/CONTRIB.md
>   https://github.com/GoogleCloudPlatform/appengine-maven-archetypes-java/pull/1/files#diff-5
>   (38)
> - _A_
>   guestbook-archetype/src/main/resources/archetype-resources/CONTRIBUTING.md
>   https://github.com/GoogleCloudPlatform/appengine-maven-archetypes-java/pull/1/files#diff-6
>   (35)
> - _D_
>   guestbook-archetype/src/main/resources/archetype-resources/eclipse-launch-profiles/DevAppServer.launch
>   https://github.com/GoogleCloudPlatform/appengine-maven-archetypes-java/pull/1/files#diff-7
>   (14)
> - _D_
>   guestbook-archetype/src/main/resources/archetype-resources/eclipse-launch-profiles/UpdateApplication.launch
>   https://github.com/GoogleCloudPlatform/appengine-maven-archetypes-java/pull/1/files#diff-8
>   (14)
> - _M_
>   guestbook-archetype/src/main/resources/archetype-resources/pom.xml
>   https://github.com/GoogleCloudPlatform/appengine-maven-archetypes-java/pull/1/files#diff-9
>   (40)
> - _A_
>   guestbook-archetype/src/main/resources/archetype-resources/src/main/java/Greeting.java
>   https://github.com/GoogleCloudPlatform/appengine-maven-archetypes-java/pull/1/files#diff-10
>   (83)
> - _A_
>   guestbook-archetype/src/main/resources/archetype-resources/src/main/java/Guestbook.java
>   https://github.com/GoogleCloudPlatform/appengine-maven-archetypes-java/pull/1/files#diff-11
>   (34)
> - _M_
>   guestbook-archetype/src/main/resources/archetype-resources/src/main/java/GuestbookServlet.java
>   https://github.com/GoogleCloudPlatform/appengine-maven-archetypes-java/pull/1/files#diff-12
>   (27)
> - _A_
>   guestbook-archetype/src/main/resources/archetype-resources/src/main/java/OfyHelper.java
>   https://github.com/GoogleCloudPlatform/appengine-maven-archetypes-java/pull/1/files#diff-13
>   (48)
> - _M_
>   guestbook-archetype/src/main/resources/archetype-resources/src/main/java/SignGuestbookServlet.java
>   https://github.com/GoogleCloudPlatform/appengine-maven-archetypes-java/pull/1/files#diff-14
>   (28)
> - _M_
>   guestbook-archetype/src/main/resources/archetype-resources/src/main/webapp/WEB-INF/appengine-web.xml
>   https://github.com/GoogleCloudPlatform/appengine-maven-archetypes-java/pull/1/files#diff-15
>   (4)
> - _M_
>   guestbook-archetype/src/main/resources/archetype-resources/src/main/webapp/WEB-INF/web.xml
>   https://github.com/GoogleCloudPlatform/appengine-maven-archetypes-java/pull/1/files#diff-16
>   (21)
> - _M_
>   guestbook-archetype/src/main/resources/archetype-resources/src/main/webapp/guestbook.jsp
>   https://github.com/GoogleCloudPlatform/appengine-maven-archetypes-java/pull/1/files#diff-17
>   (69)
> - _M_
>   guestbook-archetype/src/main/resources/archetype-resources/src/test/java/GuestbookServletTest.java
>   https://github.com/GoogleCloudPlatform/appengine-maven-archetypes-java/pull/1/files#diff-18
>   (3)
> - _M_
>   guestbook-archetype/src/main/resources/archetype-resources/src/test/java/SignGuestbookServletTest.java
>   https://github.com/GoogleCloudPlatform/appengine-maven-archetypes-java/pull/1/files#diff-19
>   (22)
> - _M_
>   guestbook-archetype/src/test/resources/projects/basic/archetype.properties
>   https://github.com/GoogleCloudPlatform/appengine-maven-archetypes-java/pull/1/files#diff-20
>   (2)
> 
> Patch Links:
> 
>    -
>    https://github.com/GoogleCloudPlatform/appengine-maven-archetypes-java/pull/1.patch
>    -
>    https://github.com/GoogleCloudPlatform/appengine-maven-archetypes-java/pull/1.diff
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/GoogleCloudPlatform/appengine-maven-archetypes-java/pull/1
> .

tmatsuo commented on appengine-skeleton-archetype/pom.xml:
```diff
@@ -18,8 +18,8 @@
-      <name>Takashi Matsuo</name>
-      <email>tmatsuo@google.com</email>
+      <name>Les Vogel</name>
+      <email>lesv@google.com</email>
```
Adding you is the right thing. I'm ok with removing me, but I think it is natural to keep both.

lesv replied:
Will do.

tmatsuo replied:
Let's do it

tmatsuo commented on guestbook-archetype/pom.xml:
```diff
@@ -18,8 +18,8 @@
   </licenses>
   <developers>
     <developer>
-      <name>Takashi Matsuo</name>
```
ditto

tmatsuo commented on guestbook-archetype/src/main/resources/META-INF/maven/archetype-metadata.xml:
```diff
@@ -29,12 +29,6 @@
-      <includes>
-        <include>**/*.launch</include>
-      </includes>
-    </fileSet>
```
Why removing?

lesv replied:
It seemed like a good idea since folks weren't using the eclipse plugin.  Ludo pointed out that these help Eclipse on MVN, so it will come back.

tmatsuo commented on guestbook-archetype/src/main/resources/archetype-resources/pom.xml:
```diff
@@ -22,7 +30,7 @@
             <groupId>com.google.appengine</groupId>
             <artifactId>appengine-api-1.0-sdk</artifactId>
-            <version>${appengine-version}</version>
+            <version>${symbol_dollar}{appengine.version}</version>
```
I think it is supposed to work. Have you tested it?

tmatsuo replied:
I tested it, unfortunately it doesn't work. See my other pull request.

lesv replied:
It worked for me.  Yes -- I tested it.

tmatsuo replied:
How did you test it?

lesv replied:
Apparently not well enough -- just did more -- crash. :(

tmatsuo commented on guestbook-archetype/src/main/resources/archetype-resources/src/main/webapp/WEB-INF/appengine-web.xml:
```diff
@@ -3,8 +3,8 @@
 <appengine-web-app xmlns="http://appengine.google.com/ns/1.0">
-    <application>${application-id}</application>
-    <version>${symbol_dollar}{appengine.app.version}</version>
+    <application>${symbol_dollar}{app.id}</application>
```
Why do you want to do this?

Edit: Oh I see, never mind. Now you have app.id and app.version in the pom file.

lesv replied:
I suspect I don't need them - I'll test again.

tmatsuo commented:
First round of comments

tmatsuo commented on guestbook-archetype/pom.xml:
```diff
@@ -4,7 +4,7 @@
   <groupId>com.google.appengine.archetypes</groupId>
   <artifactId>guestbook-archetype</artifactId>
-  <version>2.0.0-1.9.10</version>
+  <version>2.0.0-1.9.20</version>
```
Moving to objectify is a major change, how about to change the version to 3.0.0-1.9.20 ?

lesv replied:
Seems like a very long version string.  But OK.

tmatsuo commented on guestbook-archetype/src/main/resources/archetype-resources/CONTRIBUTING.md:
```diff
@@ -0,0 +1,35 @@
+# How to become a contributor and submit your own code
```
I included this file because I originally intended to use this archetype to build one of our repo (appengine-guestbook-java). However, on the second thought, since this will potentially become user's project, we should remove this file.

lesv replied:
Agree

Commit:
Try #2

1. Add gcloud plugin
2. Remove contributing as this is for the developer.
3. Add back Eclipse launch profiles
4. get rid of ${symbol_dollar} stuff in pom, turns out they aren’t
required.
5. Missed package in guestbook.jsp
6. # not required to be substituted.
7.

lesv commented:
Implement issue #5

lesv commented:
@tmatsuo PTAL

tmatsuo commented on guestbook-archetype/src/main/resources/META-INF/maven/archetype-metadata.xml:
```diff
@@ -29,11 +29,11 @@
-      <includes>
-        <include>**/*.launch</include>
-      </includes>
+    <fileSet encoding="UTF-8">		
```
What are the trailing green spaces on the diff?

tmatsuo commented:
LGTM except for small nits

Commit:
Remove tabs at EOL.

tmatsuo commented on guestbook-archetype/pom.xml:
```diff
@@ -18,11 +18,17 @@
   <developers>
     <developer>
-      <name>Takashi Matsuo</name>
+      <name>Takashi Matsou</name>
```
Matsou -> Matsuo

tmatsuo commented on guestbook-archetype/src/main/resources/archetype-resources/src/main/webapp/WEB-INF/logging.properties:
```diff
@@ -1,16 +1,13 @@
+#
 
-${symbol_pound} Set the default logging level for all loggers to WARNING
+# Set the default logging level for all loggers to WARNING
```
Is ${symbol_pound} problematic?

lesv replied:
symbol_pound was how I found the file.  I've removed them.

tmatsuo replied:
You should mark these files not to filter.

Commit:
Cleanups

1. Put Takashi back in Skeleton developers
2. Clean up skeleton & guestbook for mvn substitution.
3. Fix spelling of Takashi’s last name.

tmatsuo commented on appengine-skeleton-archetype/src/main/resources/archetype-resources/src/main/webapp/WEB-INF/appengine-web.xml:
```diff
@@ -1,10 +1,7 @@
 <appengine-web-app xmlns="http://appengine.google.com/ns/1.0">
     <application>${application-id}</application>
-    <version>${symbol_dollar}{appengine.app.version}</version>
+    <version>${appengine.app.version}</version>
```
Not a big deal, but
This file will be filtered. If you add appengine.app.version to the archetype metadata, this value will be replaced with the specified value, which I don't think the expected behavior.

tmatsuo commented on appengine-skeleton-archetype/src/main/resources/archetype-resources/src/main/webapp/WEB-INF/logging.properties:
```diff
@@ -1,16 +1,13 @@
+# </system-properties>
+#
 
-${symbol_pound} Set the default logging level for all loggers to WARNING
```
Yes, we don't need them.
It's a small thing, but if you stop using the ${symbol_xxx}, you should not filter these files. See archetype-metadata.xml for details.

lesv replied:
Actually, you'll find that if they aren't listed as requiredParameters they don't get substituted. Only the values we ask to set something get done.

tmatsuo replied:
Can you stop filtering those files?

tmatsuo commented on appengine-skeleton-archetype/src/main/resources/archetype-resources/src/main/webapp/WEB-INF/web.xml:
```diff
@@ -1,6 +1,3 @@
-#set( $symbol_pound = '#' )
-#set( $symbol_dollar = '$' )
-#set( $symbol_escape = '\' )
```
ditto

tmatsuo commented on guestbook-archetype/src/main/resources/archetype-resources/src/main/webapp/WEB-INF/appengine-web.xml:
```diff
@@ -1,10 +1,7 @@
-#set( $symbol_pound = '#' )
-#set( $symbol_dollar = '$' )
-#set( $symbol_escape = '\' )
```
ditto

tmatsuo commented on guestbook-archetype/src/main/resources/archetype-resources/src/main/webapp/WEB-INF/logging.properties:
```diff
@@ -1,16 +1,13 @@
-${symbol_pound} <system-properties>
-${symbol_pound}   <property name="java.util.logging.config.file" value="WEB-INF/logging.properties"/>
-${symbol_pound} </system-properties>
-${symbol_pound}
```
ditto

tmatsuo commented:
I just noticed that the pom.xml for appengine-skeleton-archetype still doesn't have ${appengine.version} variable. Maybe you can add it too?

Commit:
Don't filter some files

1. no need to filter nbactions.xml
2. web.xml doesn’t require #set’s

While web.xml needs to be filtered, and gb/web.xml does not.  AE-Skel
WEB-INF/*.xml needs filtering.

Commit:
Add clues that we are running in a Velocity world

1. add #set’s at front of files that have substitution going on.
(Note - mvn compile plugin appears to have it set ON)
2. don’t run substitution on .properties files.

lesv commented:
@tmatsuo PTAL

tmatsuo commented:
I just noticed that the pom.xml for appengine-skeleton-archetype still doesn't have ${appengine.version} variable and instead it has multiple ${appengine-version}. Maybe you can change it too?

Commit:
Bring ae-skeleton to new format

1. app.id & app.version
2. add gcloud plugin
3. appengine-version => appengine.version through substitution.

tmatsuo commented:
Thanks, LGTM
